### PR TITLE
Update oauth errors to use api errors

### PR DESCRIPTION
**Pull request by dmcgowan**
For more consistency in handling client errors, this just converts to an API error instead of having the caller handle yet another type of error.

### CODE REVIEW

File: registry/client/errors.go [hunk 1]
```diff
@@ -109,15 +90,25 @@ func parseHTTPErrorResponse(statusCode int, r io.Reader) error {
...
-				if errStr != "" {
-					return &OAuthError{
-						ErrorCode:   errStr,
-						Description: c.Parameters["error_description"],
-					}
+				var err errcode.Error
+				// codes defined at https://tools.ietf.org/html/rfc6750#section-3.1
+				switch c.Parameters["error"] {
+				case "invalid_token":
+					err.Code = errcode.ErrorCodeUnauthorized
```
Comment (stevvooe) [inline registry/client/errors.go]: If there is already a `DENIED`/`UNAUTHORIZED` error, can we just use that?

Comment (dmcgowan) [inline registry/client/errors.go]: They potentially have different messages, from the client perspective right now we only use the first in the list. We currently don't have a defined use of the remainder, we should probably at least log them.

**Comment by stevvooe** [comment]
LGTM

**Comment by dmcgowan** [comment]
Test with engine, much better, merging

**Pull request merged**